packages/devtools_app/lib/src/screens/memory/panes/diff/widgets/class_filter_dialog.dart

```diff
@@ -0,0 +1,159 @@
+// Copyright 2022 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import '../../../../../shared/common_widgets.dart';
+import '../../../../../shared/dialogs.dart';
+import '../../../../../shared/theme.dart';
+import '../../../../../shared/utils.dart';
+import '../../../shared/heap/class_filter.dart';
+import '../controller/utils.dart';
+
+class ClassFilterDialog extends StatefulWidget {
+  const ClassFilterDialog(this.classFilter, {super.key});
+
+  final ValueNotifier<ClassFilter> classFilter;
+
+  @override
+  State<ClassFilterDialog> createState() => _ClassFilterDialogState();
+}
+
+class _ClassFilterDialogState extends State<ClassFilterDialog> {
+  bool _initialized = false;
+  bool _showHelp = false;
+  late String _rootPackage;
+
+  late ClassFilterType _type;
+  final _except = TextEditingController();
+  final _only = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+    unawaited(_initialize());
+  }
+
+  Future<void> _initialize() async {
+    assert(!_initialized);
+    _rootPackage = await tryToDetectRootPackage() ?? '';
+    _loadStateFromFilter(widget.classFilter.value);
+    setState(() => _initialized = true);
+  }
+
+  @override
+  void didUpdateWidget(covariant ClassFilterDialog oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (oldWidget.classFilter.value != widget.classFilter.value) {
+      _loadStateFromFilter(widget.classFilter.value);
+    }
+  }
+
+  void _loadStateFromFilter(ClassFilter filter) {
+    _type = filter.filterType;
+    _except.text = filter.except;
+    _only.text = filter.only ?? _rootPackage;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+    final textFieldLeftPadding = scaleByFontFactor(40.0);
+    final itemSpacing = scaleByFontFactor(28.0);
+
+    void onTypeChanged(ClassFilterType? type) => setState(() => _type = type!);
+
+    return DevToolsDialog(
+      title: dialogTitleText(theme, 'Filter Classes and Packages'),
+      content: !_initialized
+          ? Progress(size: 30.0)
+          : Container(
+              width: defaultDialogWidth,
+              child: Column(
+                mainAxisSize: MainAxisSize.min,
+                crossAxisAlignment: CrossAxisAlignment.start,
+                children: [
+                  IconLabelButton(
+                    tooltip: _showHelp ? 'Hide help' : 'Show help',
+                    icon: _showHelp
+                        ? Icons.keyboard_arrow_up
+                        : Icons.keyboard_arrow_down,
+                    label: 'Help',
+                    onPressed: () => setState(() => _showHelp = !_showHelp),
+                  ),
+                  if (_showHelp) ...[
```

Comment:
instead of conditionally showing this help information, why don't we always show it in a subtle text style? This is one less click / thing for the user to figure out, and they'll likely need instructions and examples anyway. See example from other Filter dialogs in DevTools:
![Screen Shot 2022-10-24 at 4 23 08 PM](https://user-images.githubusercontent.com/43759233/197648454-c88a235c-23ac-4815-9073-efe793337f7a.png)